gstreamer/test/test-audio.rb

```diff
@@ -0,0 +1,36 @@
+# Copyright (C) 2024  Ruby-GNOME Project Team
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+
+class TestAudio < Test::Unit::TestCase
+  def test_audio
+    caps = Gst::Caps.new("audio/ogg")
+    caps.set_value("format", "S8")
+    caps.set_value("rate", 44100)
+    caps.set_value("channels", 2)
+
+    success, audio_info = Gst::AudioInfo.from_caps(caps)
+    assert(success)
```

Comment:
Hmm. Does `Gst::AudioInfo.new(caps)` work?

Reply:
@kou 
Yes. I committed and pushed.

How did you know that? [`gst_audio_info_new`](https://gstreamer.freedesktop.org/documentation/audio/audio-info.html?gi-language=c#gst_audio_info_new) signature doesn't indicate any argument. I want to know it to infer GObject Introspection behavior.

Reply:
gobject-introspection gem creates a constructor (`#initialize`) that dispatches underlying constructors (`gst_audio_info_new()` and `gst_audio_info_new_from_caps()` in this case) automatically.

In this case, we can distinct `gst_audio_info_new()` and `gst_audio_info_new_caps()` calls by the given arguments. So `Gst::AudioInfo.new(caps)` works.

Reply:
How flexible! Thank you for describing.